### PR TITLE
Fix add reduce

### PR DESCRIPTION
**Pull request by denolf**
* fixed Makefile
* fixed run.lit
* made input_size variable
* moved kernel to aie2 folder

### CODE REVIEW

**Comment by jamesroxbypb** [comment]
Looks good, but should remove all the stuff to do with vector_max and vector_min (the aie2.py and the .cc's) for now.  Once this is approved and merged, a separate PR should then do the same with those 2 designs.  Right now for example, the aie2.py for vector_max_reduce is failing the formatting check.

**Comment by fifield** [comment]
but it is still failing CI?
```
********************
Failed Tests (10):
  AIE_REFERENCE_DESIGNS :: basic/passthrough_kernel/run.lit
  AIE_REFERENCE_DESIGNS :: basic/reduce_add/run.lit
  AIE_REFERENCE_DESIGNS :: basic/vector_max/run.lit
  AIE_REFERENCE_DESIGNS :: basic/vector_max_reduce/run.lit
  AIE_REFERENCE_DESIGNS :: basic/vector_min/run.lit
  AIE_REFERENCE_DESIGNS :: basic/vector_min_reduce/run.lit
  AIE_REFERENCE_DESIGNS :: vision/color_detect/run.lit
  AIE_REFERENCE_DESIGNS :: vision/color_threshold/run.lit
  AIE_REFERENCE_DESIGNS :: vision/edge_detect/run.lit
  AIE_REFERENCE_DESIGNS :: vision/passthrough/run.lit
```
FYI I have[ fixes for the vision examples](https://github.com/fifield/mlir-aie/commit/fbc098e5e9381b04cf598fb9bbe5034ff357ae61) in the merge to main I've already started to prepare.

**Comment by jamesroxbypb** [comment]
Jeff, I also looked at those failing tests.  For this PR, there should only be 1 directory that is affected and its test should pass.  The other passes/fails are less relevant.

File: programming_examples/basic/reduce_add/Makefile [hunk 1]
```diff
@@ -8,25 +8,27 @@
...
+
 ACDC_AIE = $(dir $(shell which aie-opt))/..
 
-targetname = vector_max
+targetname = add_reduce
 devicename = ipu
 col = 0
 CHESS_FLAGS=${CHESSCCWRAP2_FLAGS}
-KERNEL_LIB=${ACDC_AIE}/../../aie_kernels/aie2/
+KERNEL_LIB=../../../aie_kernels/aie2/
```
Comment (eddierichter-amd) [inline programming_examples/basic/reduce_add/Makefile:19]: Does `which aiecc.py` not generate an output in your environment?

File: programming_examples/basic/reduce_add/Makefile [hunk 2]
```diff
@@ -8,25 +8,27 @@
...
 
 all: build/final.xclbin build/insts.txt
 
-build/i32_add_reduce.o: ${KERNEL_LIB}/i32_add_reduce.cc
+build/reduce_add.cc.o: ${KERNEL_LIB}/reduce_add.cc
 	mkdir -p ${@D}
-	cd ${@D} && xchesscc_wrapper ${CHESS_FLAGS} -c $< -o ${@F}
+	cd ${@D} && xchesscc_wrapper ${CHESSCCWRAP2_FLAGS} -c $(<:%=../%) -o ${@F}
```
Comment (eddierichter-amd) [inline programming_examples/basic/reduce_add/Makefile:25]: Was CHESS_FLAGS not being set correctly? It is set to default to this AIE2 on default but when building on VCK5000 it gets set to `aie`.

Comment (denolf) [reply programming_examples/basic/reduce_add/Makefile:25]: will move back to CHESS_FLAGS

**Pull request merged**